components/clarity-repl/src/repl/hooks/perf.rs

```diff
@@ -311,8 +323,83 @@ impl EvalHook for PerfHook {
+
+                            for (call_stack, cost) in &self.collected_data {
+                                let final_call_stack = format!("{};{}", contract, call_stack);
+                                writeln!(self.writer, "{} {}", final_call_stack, cost)
+                                    .expect("Failed to write to perf output");
+                            }
+                        } else {
+                            for (call_stack, cost) in &self.collected_data {
+                                writeln!(self.writer, "{} {}", call_stack, cost)
+                                    .expect("Failed to write to perf output");
+                            }
+                        }
+                    } else {
+                        for (call_stack, cost) in &self.collected_data {
+                            writeln!(self.writer, "{} {}", call_stack, cost)
+                                .expect("Failed to write to perf output");
+                        }
+                    }
+                }
+
+                #[cfg(not(target_arch = "wasm32"))]
+                {
+                    if let Some(ref contract) = self.contract_identifier {
+                        let mut file = std::fs::File::create("perf.data")
+                            .expect("Failed to create perf output file");
+
+                        if overhead_runtime > 0 {
+                            writeln!(
+                                &mut file,
+                                "{}:{} {}",
+                                contract, OVERHEAD_HEADER, overhead_runtime
+                            )
+                            .expect("Failed to write overhead to perf output");
+
+                            for (call_stack, cost) in &self.collected_data {
+                                let final_call_stack = format!("{};{}", contract, call_stack);
+                                writeln!(&mut file, "{} {}", final_call_stack, cost)
+                                    .expect("Failed to write to perf output");
+                            }
+                        } else {
+                            for (call_stack, cost) in &self.collected_data {
+                                writeln!(&mut file, "{} {}", call_stack, cost)
+                                    .expect("Failed to write to perf output");
+                            }
+                        }
+                    } else {
+                        // overhead failed for some reason, write the collected data
+                        let mut file = std::fs::File::create("perf.data")
+                            .expect("Failed to create perf output file");
+                        for (call_stack, cost) in &self.collected_data {
+                            writeln!(&mut file, "{} {}", call_stack, cost)
+                                .expect("Failed to write to perf output");
+                        }
+                    }
+                }
```

Comment:
I think we can limit some code duplication between wasm32 and no wasm32 code.
What about keeping the writer into a variable, and whole the rest can be target agnostic?

```rust
                #[cfg(not(target_arch = "wasm32"))]
                let mut writer =
                    std::fs::File::create("perf.data").expect("Failed to create perf output file");

                #[cfg(target_arch = "wasm32")]
                self.clear_buffer();
                #[cfg(target_arch = "wasm32")]
                let writer = &mut self.writer;
```


There are already difference between the two implementation (such as the order of the conditions), and I'm afraid it will be hard to maintain in the future